verification/verify.go

```diff
@@ -103,7 +112,29 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 		return nil, nil
 	}
 
-	// TODO sig checking; extract from broadcast.go
+	if params.Results != nil {
+		segHashes := make([][]byte, len(params.Results.Segments))
+		for i, segment := range params.Results.Segments {
+
+			data, err := drivers.GetSegmentData(segment.Url)
+			if err != nil {
+				glog.Errorf("%v error with segment: %v (URL: %v)", monitor.SegmentTranscodeErrorDownload, err, segment.Url)
+				break
+			}
+			segHashes[i] = crypto.Keccak256(data)
+		}
+
+		// Might not have seg hashes if results are directly uploaded to the broadcaster's OS
+		// TODO: Consider downloading the results to generate seg hashes if results are directly uploaded to the broadcaster's OS
+		if len(segHashes) != len(params.Results.Segments) &&
```

Comment:
If we exit early with an error when we cannot download the segment data above then we don't need this check.

Reply:
Agreed - we don't need *this* check, but we do still need a check.   After reviewing 4def93a69fdbc03ac0623ab0ece10451070649a9 , I think you'll agree with me. I changed the check from `len(segHashes) != len(params.Results.Segments)` to `len(segHashes) != len(params.Renditions)`